baseplate/core.py

```diff
@@ -132,305 +136,177 @@ def from_upstream(cls, trace_id, parent_id, span_id, sampled, flags):
+    "_User", ["authentication_token", "loid", "cookie_created_ms"])
 _OAuthClient = collections.namedtuple(
-    "_OAuthClient", ["authentication_context"])
+    "_OAuthClient", ["authentication_token"])
 Session = collections.namedtuple("Session", ["id"])
 
 
 class User(_User):
-    """Wrapper for the user values in AuthenticationContext and the LoId cookie.
+    """Wrapper for the user values in AuthenticationToken and the LoId cookie.
     """
 
-    def event_fields(self):
-        """Dictionary of values to be added to events in the current context
-        """
-        if self.is_logged_in:
-            user_id = self.id
-        else:
-            user_id = self.loid
-        return {
-            "user_id": user_id,
-            "user_logged_in": self.is_logged_in,
-            "cookie_created": self.cookie_created_ms,
-        }
-
     @property
     def id(self):
-        """Authenticated account_id for the current User
+        """Authenticated account_id for the current User.
 
         :type: account_id string or None if context authentication is invalid
-        :raises: :py:class:`UndefinedSecretsException` if the
-            :py:class:`SecretsStore` has not been bound to the context handling
-            class (means that the authentication payload could not be decrypted
-            and the user_id returned)
-        :raises: :py:class:`WithheldAuthenticationError` if there was no
-            authentication token defined for the current context
+        :raises: :py:class:`NoAuthenticationError` if there was no
+            authentication token, it was invalid, or the subject is not an
+            account.
 
         """
-        return self.authentication_context.account_id
+        subject = self.authentication_token.subject
+        if not subject.startswith("t2_"):
+            raise NoAuthenticationError
+        return subject
 
     @property
     def is_logged_in(self):
-        """Does the User have a valid, authenticated id"""
+        """Does the User have a valid, authenticated id?"""
         try:
             return self.id is not None
-        except (WithheldAuthenticationError, UndefinedSecretsException):
+        except NoAuthenticationError:
             return False
 
     @property
     def roles(self):
-        """Authenticated roles for the current User
+        """Authenticated roles for the current User.
 
         :type: set(string)
-        :raises: :py:class:`UndefinedSecretsException` if the
-            :py:class:`SecretsStore` has not been bound to the context handling
-            class (means that the authentication payload could not be decrypted
-            and the user_id returned)
-        :raises: :py:class:`WithheldAuthenticationError` if there was no
-            authentication token defined for the current context
+        :raises: :py:class:`NoAuthenticationError` if there was no
+            authentication token or it was invalid
```

Comment:
added a `has_role`. since roles are used for security, I felt it'd be safer to not do a plural version since the interface there gets kinda confusing.